plugins/osdn/vnids.go

```diff
@@ -72,194 +111,37 @@ func (oc *OsdnController) unSetVNID(name string) (id uint, err error) {
-				idSet.Insert(int(id))
-			}
-		}
-	}
-	return ids
-}
-
 func populateVNIDMap(oc *OsdnController) error {
-	nets, err := oc.Registry.GetNetNamespaces()
+	nsList, err := oc.Registry.GetNamespaces()
 	if err != nil {
 		return err
 	}
 
-	for _, net := range nets {
-		oc.setVNID(net.Name, net.NetID)
-	}
-	return nil
-}
-
-func (oc *OsdnController) VnidStartMaster() error {
-	err := populateVNIDMap(oc)
-	if err != nil {
-		return err
-	}
-
-	// VNID: 0 reserved for default namespace and can reach any network in the cluster
-	// VNID: 1 to 9 are internally reserved for any special cases in the future
-	oc.netIDManager, err = netutils.NewNetIDAllocator(10, MaxVNID, oc.getAllocatedVNIDs())
-	if err != nil {
-		return err
-	}
-
-	// 'default' namespace is currently always an admin namespace
-	oc.adminNamespaces = append(oc.adminNamespaces, "default")
-
-	go utilwait.Forever(oc.watchNamespaces, 0)
-	return nil
-}
-
-func (oc *OsdnController) isAdminNamespace(nsName string) bool {
-	for _, name := range oc.adminNamespaces {
-		if name == nsName {
-			return true
-		}
-	}
-	return false
-}
-
-func (oc *OsdnController) assignVNID(namespaceName string) error {
-	// Nothing to do if the netid is in the vnid map
-	if _, err := oc.GetVNID(namespaceName); err == nil {
-		return nil
-	}
-
-	// If NetNamespace is present, update vnid map
-	netns, err := oc.Registry.GetNetNamespace(namespaceName)
-	if err == nil {
-		oc.setVNID(namespaceName, netns.NetID)
-		return nil
-	}
-
-	// NetNamespace not found, so allocate new NetID
-	var netid uint
-	if oc.isAdminNamespace(namespaceName) {
-		netid = AdminVNID
-	} else {
-		var err error
-		netid, err = oc.netIDManager.GetNetID()
+	for _, ns := range nsList {
+		id, err := netid.GetVNID(&ns)
 		if err != nil {
-			return err
-		}
-	}
-
-	// Create NetNamespace Object and update vnid map
-	err = oc.Registry.WriteNetNamespace(namespaceName, netid)
-	if err != nil {
-		e := oc.netIDManager.ReleaseNetID(netid)
-		if e != nil {
-			log.Errorf("Error while releasing netid: %v", e)
-		}
-		return err
-	}
-	oc.setVNID(namespaceName, netid)
-	return nil
-}
-
-func (oc *OsdnController) revokeVNID(namespaceName string) error {
-	// Remove NetID from vnid map
-	netid_found := true
-	netid, err := oc.unSetVNID(namespaceName)
-	if err != nil {
-		log.Error(err)
-		netid_found = false
-	}
-
-	// Delete NetNamespace object
-	err = oc.Registry.DeleteNetNamespace(namespaceName)
-	if err != nil {
-		return err
-	}
-
-	// Skip NetID release if
-	// - Value matches AdminVNID as it is not part of NetID allocation or
-	// - NetID is not found in the vnid map
-	if (netid == AdminVNID) || !netid_found {
-		return nil
-	}
-
-	// Check if this netid is used by any other namespaces
-	// If not, then release the netid
-	if !oc.checkVNID(netid) {
-		err = oc.netIDManager.ReleaseNetID(netid)
-		if err != nil {
-			return fmt.Errorf("Error while releasing netid %d for namespace %q, %v", netid, namespaceName, err)
+			continue
```

Comment:
since I'm guessing this will show a useless code snippet, I'm commenting on:

    func populateVNIDMap(oc *OsdnController) error {
        ...
        id, err := netid.GetVNID(&ns)
        if err != nil { 
            continue

This ignores both "annotation isn't set" and "annotation is set, but to a nonsense-value", which we probably ought to warn about